core/includes/file.inc

```diff
@@ -1832,6 +1833,48 @@ function file_validate_image_resolution(File $file, $maximum_dimensions = 0, $mi
   return $errors;
 }
 
+/**
+ * Check for EXIF orientation data and rotate file if needed.
+ *
+ * @param File $file
+ *   A file entity.
+ * @param $orientation
+ *   A flag indicating that EXIF data should be checked and image rotated if necessary.
+ */
+function file_validate_image_orientation( $file, $orientation) {
```

Comment:
This function needs to more closely match the other file_validate_* functions to prevent errors.

It's expecting an array of errors to be returned. It's not clear what error to return here so it can just return an empty array I think. 

There should also be some checking that it's a file and an image. I copied some of this from file_validate_image_resolution():

```
/**
 * Check for EXIF orientation data and rotate file if needed.
 *
 * @param File $file
 *   A file entity.
 * @param $orientation
 *   A flag indicating that EXIF data should be checked and image rotated if necessary.
 *
 * @return
 *   An array. If the file is an image and did not meet the requirements, it
 *   will contain an error message.
 *
 * @see hook_file_validate()
 */
function file_validate_image_orientation(File $file, $orientation = 0) {
  $errors = array();

  // Check first that the file is an image.
  if ($info = image_get_info($file->uri)) {
    // Is orientation required and is PHP function available?
    if ($orientation && function_exists('exif_read_data')) {
      $file_exif = @exif_read_data(backdrop_realpath($file->uri));
      // If the Orientation key|value exists rotate as required.
      if (is_array($file_exif) && isset($file_exif['Orientation'])) {
        // Orientation numbers and corresponding degrees.
        // @note: Odd numbers are flipped images, would need different process.
        switch ($file_exif['Orientation']) {
          case 3:
            $degrees = 180;
            break;
          case 6:
            $degrees = 90;
            break;
          case 8:
            $degrees = 270;
            break;
          default:
            $degrees = 0;
        }
        if ($degrees > 0) {
          // Load the image object for manipulation.
          $file_image = image_load($file->uri);
          if (image_rotate($file_image, $degrees, 0xffffff)) {
            image_save($file_image);
          }
          backdrop_set_message(t('The image was rotated by @degrees degrees.', array('@degrees' => $degrees)));
        }
      }
    }
  }

  return $errors;
}
```
Note that the parameter has `File $file` as it's expecting an object of that class. And it's wrapped in an if statement to check that it's an image.